### PR TITLE
fix: hide workspace menu options trigger when there are no permissions

### LINKED ISSUE #18743: [Bug]: When the user have no access to any of the options in workspace menu options, hide the trigger.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When the user has no access to perform any operations in workspace we should hide the workspace menu options trigger, the more menu icon.

### Steps To Reproduce

1. Remove all workspace permissions for the user
2. Go to homepage
3. Click on workspace menu options icon
4. Doesn't do anything.

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

RBAC DP

### PR DESCRIPTION
**Pull request by berzerkeer**
## Description

When the user have no access to any of the options under workspace menu options, simply hide the trigger.

Fixes #18743 
> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/client/src/pages/Applications/index.tsx [hunk 1]
```diff
@@ -705,7 +705,7 @@ function ApplicationsSection(props: any) {
 
         const showWorkspaceMenuOptions =
           canInviteToWorkspace ||
-          hasManageWorkspacePermissions ||
+          hasCreateNewApplicationPermission ||
```
Comment (ankitakinger) [inline app/client/src/pages/Applications/index.tsx:708]: shouldn't this be checking both the conditions?

Comment (berzerkeer) [reply app/client/src/pages/Applications/index.tsx:708]: Sorry my bad , meant to replace the other one. Let me update

**Review by ankitakinger — APPROVED** [review]

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**